### PR TITLE
Replace custom resource functions with generated clientset

### PR DESCRIPTION
**Pull request by cbandy**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [x] Have you updated or added documentation for the change, as applicable?
 - [x] Have you tested your changes on all related environments with successful results, as applicable?

The first commit is mostly a mechanical transformation of the following patterns:

```diff
- var result crv1.X
- found, err := kubeapi.GetX(RESTClient, &result, name, namespace)
+ result, err := clientset.CrunchydataV1().X(namespace).Get(name, metav1.GetOptions{})
```

```diff
- var list crv1.XList
- err := kubeapi.GetXBySelector(RESTClient, &list, selector, namespace)
+ list, err := clientset.CrunchydataV1().X(namespace).List(metav1.ListOptions{LabelSelector: selector})
```

There are variations when it comes to the `found` boolean. Sometimes the equivalent is `kerrors.IsNotFound(err)` while other times it just `err != nil`. Once or twice a `Get()` followed by a `Delete()` is replaced with just a `Delete()`.

The second commit is optional. It targets all those places where we pass `*rest.Config`, `*rest.RESTClient`, and `kubernetes.Interface` together and replaces them with a single `*kubeapi.Client` or `kubeapi.Interface`. Those types are implemented in `internal/kubeapi/client_config.go`.

Issue: [ch8733]

### CODE REVIEW

File: internal/apiserver/root.go [hunk 1]
```diff
@@ -46,14 +45,10 @@ const PGOSecretName = "pgo.tls"
...
-	// PGOClientset is a client for PostgreSQL Operator resources
-	PGOClientset pgov1.Interface
-	// Clientset is a client for native Kubernetes resources
-	Clientset kubernetes.Interface
+	// Clientset is a client for Kubernetes resources
+	Clientset kubeapi.Interface
 	// RESTConfig holds the REST configuration for a Kube client
 	RESTConfig *rest.Config
-	// RESTClient is a REST client for the Kubernetes API
-	RESTClient *rest.RESTClient
```
Comment (cbandy) [inline internal/apiserver/root.go:56]: Here's where the clients changed in apiserver. They are initialized in the last hunk of this same file.

File: pgo-rmdata/rmdata/types.go [hunk 2]
```diff
@@ -18,14 +18,11 @@ limitations under the License.
...
-	"k8s.io/client-go/kubernetes"
-	"k8s.io/client-go/rest"
+	"github.com/crunchydata/postgres-operator/internal/kubeapi"
 )
 
 type Request struct {
-	RESTConfig       *rest.Config
-	RESTClient       *rest.RESTClient
-	Clientset        kubernetes.Interface
+	Clientset        kubeapi.Interface
```
Comment (cbandy) [inline pgo-rmdata/rmdata/types.go:25]: Here's where the clients changed in rmdata. They are initialized in `pgo-rmdata.go`.

File: pgo-scheduler/pgo-scheduler.go [hunk 3]
```diff
@@ -53,7 +53,7 @@ var namespace string
 var pgoNamespace string
 var timeout time.Duration
 var seconds int
-var kubeClient kubernetes.Interface
+var clientset kubeapi.Interface
```
Comment (cbandy) [inline pgo-scheduler/pgo-scheduler.go:56]: This is one change to the client in scheduler. It is initialized in the next hunk in this same file.

File: pgo-scheduler/scheduler/types.go [hunk 4]
```diff
@@ -18,14 +18,11 @@ package scheduler
...
-	"k8s.io/client-go/kubernetes"
-	"k8s.io/client-go/rest"
-
+	"github.com/crunchydata/postgres-operator/internal/kubeapi"
 	cv2 "github.com/robfig/cron"
 )
 
-var kubeClient kubernetes.Interface
-var restClient *rest.RESTClient
+var clientset kubeapi.Interface
```
Comment (cbandy) [inline pgo-scheduler/scheduler/types.go:25]: This is another change to the clients in scheduler. It is initialized in `scheduler.go`.

File: internal/controller/manager/controllermanager.go [hunk 5]
```diff
@@ -232,75 +229,60 @@ func (c *ControllerManager) addControllerGroup(namespace string) error {
...
+	client, err := kubeapi.NewClient()
 	if err != nil {
 		log.Error(err)
 		return err
 	}
 
-	config := clients.Config
-	pgoClientset := clients.PGOClientset
-	pgoRESTClient := clients.PGORestclient
-	kubeClientset := clients.Kubeclientset
```
Comment (cbandy) [inline internal/controller/manager/controllermanager.go:244]: Here's where the clients changed in postgres-operator.

File: pgo-rmdata/rmdata/process.go [hunk 6]
```diff
@@ -70,8 +68,9 @@ func Delete(request Request) {
 			log.Error(err)
 		}
 		//delete the pgreplica CRD
-		if err = kubeapi.Deletepgreplica(request.RESTClient, request.ReplicaName,
-			request.Namespace); err != nil {
+		if err = request.Clientset.
```
Comment (jkatz) [inline pgo-rmdata/rmdata/process.go:71]:
```suggestion
		if err := request.Clientset.
```

**Comment by cbandy** [comment]
Rebased and incorporated feedback.

**Pull request merged**